frontend/packages/pipelines-plugin/src/components/quicksearch/PipelineQuickSearchVersionDropdown.tsx

```diff
@@ -39,38 +41,46 @@ const PipelineQuickSearchVersionDropdown: React.FC<PipelineQuickSearchVersionDro
     }
     return acc;
   }, {});
-  return (
-    <DropdownDeprecated
+
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle
+      className="opp-quick-search-details__version-dropdown"
+      onClick={toggleIsOpen}
+      isExpanded={isOpen}
+      ref={toggleRef}
+      isDisabled={versions.length === 1}
       data-test="task-version"
+    >
+      {versionItems[selectedVersion]}
+    </MenuToggle>
+  );
+
+  return (
+    <Select
       className="opp-quick-search-details__version-dropdown"
-      dropdownItems={Object.keys(versionItems).map((key) => (
-        <DropdownItemDeprecated
-          component="button"
-          key={key}
-          label={versionItems[key]}
-          onClick={(e) => {
-            e.stopPropagation();
-            onChange(key);
-            setOpen(false);
-          }}
-        >
-          <div className="opp-quick-search-details__version-dropdown-item">
-            {versionItems[key]}
-            {isSelectedVersionInstalled(item, key) && <CheckCircleIcon color={greenColor.value} />}
-          </div>
-        </DropdownItemDeprecated>
-      ))}
       isOpen={isOpen}
-      toggle={
-        <DropdownToggleDeprecated
-          isDisabled={versions.length === 1}
-          data-test="task-version-toggle"
-          onToggle={toggleIsOpen}
-        >
-          {versionItems[selectedVersion]}
-        </DropdownToggleDeprecated>
-      }
-    />
+      onSelect={(_, value: string) => {
+        if (value) {
+          onChange(value);
+        }
+        setOpen(false);
+      }}
+      toggle={toggle}
+      onOpenChange={(open) => setOpen(open)}
+    >
+      <SelectList>
+        {Object.keys(versionItems).map((key) => (
+          <SelectOption key={key} label={versionItems[key]} value={key}>
```

Comment:
Can we have prop `isSelected` here as well for consistency, it will add a check icon to the selected option.

```suggestion
          <SelectOption key={key} label={versionItems[key]} value={key} isSelected={selectedVersion === key}>
```

Reply:
I didn't add it with this PR because it may be confused with the "installed" checkmark

Reply:
![image](https://github.com/user-attachments/assets/e38fa543-e07d-4daa-b56c-1ac44cc2f8fe)

Reply:
Ok, no need to add then, we can go ahead with it.